File: LarsV2.App/Models/Entities/Subject.cs
using Newtonsoft.Json;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace LarsV2.Models.Entities
{
    public class Subject
    {
        [Key]
        public int Id { get; set; }
        [Required(ErrorMessage = "Subject name is required")]
        [MaxLength(255)]
        public string Title { get; set; }
        public string Education { get; set; }
        public string Description { get; set; }
        public ICollection<LecturerSubject> LecturerSubjects { get; set; }
        public ICollection<Course> Courses { get; set; }
    }
}

File: LarsV2.App/ClientApp/src/app/DTO/subject.ts
export interface iSubject{
	id: number | null;
	title: string;
	education: string;
	description: string;
}

export class Subject implements iSubject {
	id: number | null;
	title: string;
	education: string;
	description: string;
	
	constructor(id: number | null, title: string, education: string, description: string) {
		this.id = id;
		this.title = title;
		this.education = education;
		this.description = description;
	}
}

export class NullSubject extends Subject {
	constructor(){
		super(null, "", "", "");
	}
}